tests/test_time_series.py

```diff
@@ -1157,6 +1157,49 @@ def test_interpolate_with_ep(self, tsd):
         tsd2 = tsd.interpolate(ts, ep)
         assert len(tsd2) == 0
 
+    def test_gradient(self, tsd):
+        """
+        Test the derivative calculation for a time series, verifying
+        correctness for a function with known derivative (sine wave).
+        """
+        times = np.arange(0, 10, 0.001)
+
+        # Tsd
+        data = np.sin(times)
+        tsd = nap.Tsd(t=times, d=data)
+        expected_gradient = np.cos(times)  # Derivative of sin(x) is cos(x)
+        gradient = tsd.gradient()
+        np.testing.assert_allclose(gradient.values, expected_gradient, atol=1e-3)
+
+        # Tsdframe
```

Comment:
This should be in `class TsdFrame` in test_time_series.py

Reply:
https://github.com/pynapple-org/pynapple/pull/451/commits/64946202de18f837247789d28f6c07ee519767c8